cmd/kubeadm/app/apis/kubeadm/v1alpha1/types.go

```diff
@@ -48,6 +48,11 @@ type MasterConfiguration struct {
 	APIServerCertSANs []string `json:"apiServerCertSANs"`
 	// CertificatesDir specifies where to store or look for all required certificates
 	CertificatesDir string `json:"certificatesDir"`
+
+	// ImageRepository what container registry to pull control plane images from
+	ImageRepository string `json:"imageRepository"`
+	// SpecialControlPlaneImage specifies if a specific container image should be used for all control plane components
+	SpecialControlPlaneImage string `json:"specialControlPlaneImage"`
```

Comment:
Idk, thought it might be better to be generic. It doesn't have to be hyperkube, you can bundle all control plane components in one image as well...

What about `HyperkubeImage` or `ControlPlaneHyperkubeImage` or `UseHyperkubeImageForControlPlane` or something else instead?

Naming is hard ;)